service-provider.md

```diff
@@ -3,4 +3,113 @@ description: Service Bindings for Service Providers
 permalink: /service-provider/
 ---
 
-# TODO
+Service providers expose bindings through a `Secret` resource with data required for connectivity.  The specification's [Provisioned Service section][provisioned-service] describes this in detail.  Alternatively, the specification also supports [Direct Secret Reference][direct-secret-reference].  The only requirement for Direct Secret Reference is a `Secret` resource with data required for connectivity.  Alternatively, if you are creating a Provisioned Service (preferred approach), you also need a custom resource with `.status.binding.name` attribute pointing to the `Secret` resource.
+
+Here is an example custom resource:
+
+```yaml
+apiVersion: example.dev/v1beta1
+kind: Database
+metadata:
+  name: database-service
+...
+status:
+  binding:
+    name: production-db-secret
+```
+
+
+{% include note.html content="Implementing the Provisioned Service contract frees the `ServiceBinding` from needing to know about the name of the `Secret` holding the credentials.  The service can update the secret name exposed over time.  This behavior not only decouples the ServiceBinding and the workload from needing to know the name of the Secret, it also enables use of immutable secrets.  An immutable secret cannot be updated after it is created.  To rotate the credentials, a new secret can be created and update the same in the Provisioned Service resource's `.status.binding.name` attribute." %}
+
+In the previous example, `production-db-secret` is the name of the `Secret` resource with data entries required for connectivity.  The `Secret` resource should contain a `type` entry that can be used for identifying the service.  It helps the application to identify the service as a relational database, key-value store, or a cache server.  There is no standardization on the value for `type`, but you can see some good examples in the [Spring Cloud Bindings][spring-cloud-bindings].  A few examples:
+
+- `postgresql`
+- `mysql`
+- `redis`
+- `mongodb`
+
+The `Secret` resource's [`type` field](https://kubernetes.io/docs/concepts/configuration/secret/#secret-types) should have a special value based on the `type` data entry value.  The expected format is `servicebinding.io/{type}`.  For example, if the value for `type` data entry is `mysql`, then the `Secret` resource's `type` field value should be `servicebinding.io/mysql`.  This recommendation helps to query `Secret` resources of particular type using field-selector. For example:
```

Comment:
it's worth adding a caveat that the type is encouraged but not mandatory, and has no runtime effect. If the secret has a different type that provides significant meaning, that value should be used instead.

Reply:
@baijum did this get addressed?